arbnode/espresso_delayed_message_fetcher.go

```diff
@@ -334,6 +334,12 @@ func (d *DelayedMessageFetcher) getDelayedMessagesInRange(ctx context.Context, b
 			continue
 		}
 
+		if seqNum > lastDelayedMessageIndex+1 {
```

Comment:
I was thinking this could be merged with the above if statement to become something a bit cleaner to read.
```suggestion
		if seqNum != lastDelayedMessageIndex+1 {
		  handle incorrect delayed message number
		}
```
However, we have to do 2 separate things, continue or stop the process based on these conditions, so I think that it wouldn't even make things less complicated tbh